src/schema/v2/home/results.ts

```diff
@@ -157,14 +157,23 @@ const moduleResults: HomePageArtworkModuleResolvers<any> = {
       })
     })
   },
-  recently_viewed_works: ({ meLoader, artworksLoader }) => {
+  recently_viewed_works: async ({ meLoader, artworksLoader }) => {
     if (!meLoader) return null
-    return meLoader().then(({ recently_viewed_artwork_ids }) => {
+    return meLoader().then(async ({ recently_viewed_artwork_ids }) => {
       if (recently_viewed_artwork_ids.length === 0) {
         return []
       }
       const ids = recently_viewed_artwork_ids.slice(0, RESULTS_SIZE)
-      return artworksLoader({ ids })
+      const artworks = await artworksLoader({
+        ids,
+      })
```

Comment:
Artworks loader returns the artworks in a different order even when `ids` and no sorting is specified. I'll check if I can fix this somehow on Gravity later.

Reply:
I'm wondering why not investigate Gravity directly? It looks as though this PR _may_ have actually resulted in some bugs + crashes on Force re: https://github.com/artsy/force/pull/12065#issuecomment-1464053723

Gravity is very straightforward to debug and we shouldn't be deferring 'real' upstream fixes in favor of downstream patches. I'm happy to help debug Gravity if there's some difficulty.